spec/features/applications/show_spec.rb

```diff
@@ -0,0 +1,220 @@
+      # When I visit an application's show page
+      visit "/applications/#{@application_4.id}"
+      
+      # And I have added one or more pets to the application
+      within".find_pet" do 
+       fill_in(:add_pet_name, with: "Rocky")
+       click_button("Submit")
+      end
+    
+      within".found_pet" do
+        within"#pet-#{@pet_3.id}" do
+          click_button("Adopt this Pet")
+        end
+    
+        within"#pet-#{@pet_1.id}" do
+          click_button("Adopt this Pet")
+        end
+      end
+
+      # Then I see a section to submit my application
+      within '.submit_application' do
+        expect(page).to have_content("Submit My Application")
+
+        # And in that section I see an input to enter why 
+        # I would make a good owner for these pet(s)
+        # When I fill in that input
+        fill_in(:adopting_reason, with: "I love animals")
+
+        # And I click a button to submit this application
+        click_button("Submit")
+      end
+    
+      # Then I am taken back to the application's show page
+      expect(current_path).to eq("/applications/#{@application_4.id}")
+
+      # And I see an indicator that the application is "Pending"
+      expect(page).to have_content("Pending")
+
+      # And I see all the pets that I want to adopt
+      within '.pets_in_application' do
+        expect(page).to have_content("Rocky")
+        expect(page).to have_content("Lucille Bald")
+      end
+
+      # And I do not see a section to add more pets to this application
+      expect(page).to_not have_content("Add a Pet to this Application")
+    end 
+
+    #User story 7 No Pets on an Application
+    it 'doesnt show a section to submit an application' do
+
+      # When I visit an application's show page
+      visit "/applications/#{@application_4.id}"
+      # And I have not added any pets to the application
+      within '.pets_in_application' do
+        expect(page).to_not have_content("Rocky")
+        expect(page).to_not have_content("Lucille Bald")
+      end
+   
+      # Then I do not see a section to submit my application
+      within '.submit_application' do
+        expect(page).to_not have_button("Submit")
+      end
+    end
+
+    # User story 8 Partial Matches for Pet Names
+    it 'displays any pet whose name PARTIALLY matches my search' do
+      # When I visit an application show page
+      visit "/applications/#{@application_1.id}"
+
+      # And I search for Pets by name 
+      # ex: "fluff", my search would match pets with names "fluffy", "fluff", and "mr. fluff"
+
+      within ".find_pet" do
+        fill_in(:add_pet_name, with: "fluff")
+        click_button("Submit")
+      end
+
+      within ".found_pet" do 
+        # Then I see any pet whose name PARTIALLY matches my search
+        expect(page).to have_content("fluffy")
+        expect(page).to have_content("fluff")
+        expect(page).to have_content("Mr. fluff")
+      end
+    end
```

Comment:
Another great test suite for this US, I don't have any suggestions for your tests. I think they are great.